### ISSUE TITLE
Create documentation that lists topics that we knowingly deferred from v1 until later

### ISSUE THREAD

**Issue by bieryAtFnal** [DUNE-DAQ/appfwk#57]
In the interest of time, some things were deferred from being included in v1.  It would probably be helpful to include a list of such topics in the v1 release notes, or in a README file that is distributed with the repo, or both, so that when people look at the code, they can understand that some choices were made because they were 'good enough' for v1, and they will be the topic of further discussion and investigation after v1 is released.

**Comment by bieryAtFnal** [comment]
Topics to be included in the document:  FSM, querying the state of a DAQProcess

**Comment by bieryAtFnal** [comment]
(not immediate) Question:  is further work on the command registering something that should be described?

**Comment by mroda88** [comment]
We have to specify in the document that `DAQModule::init()` is a temporary solution until a proper configuration manager will be in place. The role of init()  - setter for all those parameters that are not reconfigurable - will be described in the DAQModule interface Wiki

**Comment by eflumerf** [comment]
In today's meeting, we mentioned that Queue occupancy might be a metric we want to report to CCM monitoring. For the near term, we could also publish it to TRACE in the push/pop method implementations.

**Comment by alessandrothea** [comment]
Logging at every push and pop doesn’t seem a great idea.

**Comment by philiprodrigues** [comment]
Another issue we've decided to defer is modifying the cmake build method so that appfwk builds as a "normal" cmake package (outside of a superproject, with a `make install` target, etc)

**Comment by bieryAtFnal** [comment]
Adding this topic that I found in my notes:  the queue interface has can_push() and. can_pop() methods.  Phil (and possibly others) have pointed out that these methods have limited utility in a multi-threaded app with multi-producer or multi-consumer queues.  
In addition, when I was writing the _listrev_ example DAQModules, I didn't really consider using the can_push/pop methods.  Why would I bother writing retry&sleep code for can_push/pop when the full push() and pop() methods with a timeout do that for me?  
So, I wonder if it would be worthwhile to understand whether the can_push/pop methods are useful.

**Comment by brettviren** [comment]
Somewhere else on this github project which I can't find now I mentioned the need to timeout-wait for gaining the ability to receive input and/or produce output and also to act on that ability with code that is separated from the code used to gain the ability.

In some already invented wheels this is done with a separate "poller" object that takes any number of sources or sinks (generally here "sockets") and a mask that says to wait for input, for output or for both specified separately for each given socket and then returns with the subset of sockets ready for the requested action or returns with an indication of timeout.

So, if `can_push()` and `can_pop()` took a timeout, they could be narrowly useful for single-minded user code where waiting on exactly one socket is needed.  This still leaves open how the appfmk will provide the more general case.

Without this timeout-wait, I don't see any point in these methods and I think they are actively harmful as they encourage wrong user code design.
